lib/apiservers/engine/backends/container.go

```diff
@@ -57,6 +59,12 @@ import (
 type Container struct {
 }
 
+type volumeFields struct {
```

Comment:
I'm surprised you aren't getting vet/lint warnings on this about "volume" stuttering.  I suggest removing "Volume" from the member names.  Also consider renaming the struct to something like `volumeParams`.  Your call, just suggestions.